app/views/concerns/entity_displaying_view.rb

```diff
@@ -217,7 +206,7 @@ def entity_death
 
   def entity_date(date)
     # Just grab the first date in the array if present.
-    date && date.is_a?(Array) && date.length && date.first.is_a?(String) ? date.first : nil
+    date && date.is_a?(Array) && date.length && date.first.is_a?(String) ? date_eras_gregorian(date.first) : nil
```

Comment:
We should not just use the first string value of an array but scan the array for the first string value having a format supported by `#date_eras_gregorian`, use that if present, else use the first value in the array.